### ISSUE TITLE
Using setState() width not updating

### ISSUE THREAD

**Issue by fabrifls** [wwf410918701/react-textarea-auto-witdth-height#40]
I have a simple case, a component that has a state that when clicking on a button updates the value
```
export const Component = () => {
    const [test, setTest] = useState<string>("123456789");

    return (
        <>
            <AutoTextArea
                value={test}
            />
            <Button
                onClick={() => {
                    setTest("123")
                }}>
                accion
            </Button>
        </>
    );
};
```
In the first render this html is generated:
![image](https://github.com/wwf410918701/react-textarea-auto-witdth-height/assets/38505756/3027ac70-04f8-49fc-ade8-3273577f4548)

Now, when updating the value (by clicking on the button), only the **texarea** is updated, leaving the **div** with the previous value. In the view you obviously see the correct value because the one contained in the textarea is used. The problem with this is that the width is not updated, leaving it larger than it is.
![image](https://github.com/wwf410918701/react-textarea-auto-witdth-height/assets/38505756/98d901c1-4397-4370-8b8a-05dfa62c574b)

**Comment by fabrifls** [comment]
Adding this solves the problem:
`  React.useEffect(() => {
    setContent(value);
  }, [value]);`
  
  full component: 
  _`auto-width-wrapper.tsx`_
  ```
import * as React from "react";
import * as styles from "./auto-width-wrapper.css";

export interface AutoWidthWrapperProps
  extends React.HTMLAttributes<HTMLDivElement> {
  /** Update content when use input, to ensure width holder hold the
   * corrct width
   */
  renderTextArea: (onContentUpdate: (text: string) => void) => React.ReactNode;
  value?: string;
  minRows?: number;
}

/**
 * Wrapper to handle the auto width grow
 */
export const AutoWidthWrapper = ({
  renderTextArea,
  value = "",
  className: _className,
  style: _style,
  ...rest
}: AutoWidthWrapperProps) => {
  // Need to have text to hold the height
  const [content, setContent] = React.useState(value);

  // Sync content state with value prop whenever value changes
  React.useEffect(() => {
    setContent(value);
  }, [value]);

  return (
    <div className={styles.root} {...rest}>
      <div className={styles.widthHolder}>
        {/* To take up correct space when users input multiple lines */}
        {content.split("\n").map(function (item, index) {
          if (item.length === 0) {
            return (
              <React.Fragment key={`${index}-${item}`}>
                <div>{item}</div>
                <br />
              </React.Fragment>
            );
          }

          return <div key={`${index}-item`}>{item}</div>;
        })}
      </div>
      {renderTextArea((text: string) => {
        setContent(text);
      })}
    </div>
  );
};
```

**Comment by wwf410918701** [comment]
Hey fabricioloupias, thanks for your effort and will merge the bug fix shortly. And just for your reference, React recommend another way to update child state when prop changes :) 
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

**Comment by wwf410918701** [comment]
Fixed in version 1.0.3
